res/controllers/Reloop-Digital-Jockey-2-IE.scripts.js

```diff
@@ -0,0 +1,878 @@
+        if (isTraxModeEnabled) {
+            if (this.jogMode !== RDJ2.JOGMODES.trax) {
+                this.lastNonTraxJogMode = this.jogMode;
+                this.jogMode = RDJ2.JOGMODES.trax;
+                this.input = this.inputTrax;
+                //set all LEDs on to indicate trax mode
+                if (midi.sendShortMsg) {
+                    midi.sendShortMsg(0x90, this.searchMidiCtrl, RDJ2.MIDI_BUTTON_ON);
+                    midi.sendShortMsg(0x90, this.scratchMidiCtrl, RDJ2.MIDI_BUTTON_ON);
+                    midi.sendShortMsg(0x90, this.fxDryWetMidiCtrl, RDJ2.MIDI_BUTTON_ON);
+                }
+            }
+        } else {
+            this.jogMode = this.lastNonTraxJogMode;
+            this.input = this.inputNormal;
+            this.updateControls();
+        }
+    },
+    unshift: function() {
+        var isLibraryModeEnabled = engine.getValue("[Master]", "maximize_library");
+        if (!isLibraryModeEnabled) {
+            this.setTraxMode(false);
+        }
+    },
+    shift: function() {
+        var isLibraryModeEnabled = engine.getValue("[Master]", "maximize_library");
+        if (!isLibraryModeEnabled) {
+            this.setTraxMode(true);
+        }
+    },
+});
+
+/* Jog Wheel */
+
+RDJ2.Deck.prototype.onJogTouch = function(channel, control, value) {
+    var currentJogMode =  this.jogModeSelector.getJogMode();
+    this.jogTouchState = RDJ2.isButtonPressed(value);
+
+    if (currentJogMode === RDJ2.JOGMODES.vinyl && this.jogTouchState) {
+        engine.scratchEnable(this.number,
+            RDJ2.JOG_RESOLUTION,
+            RDJ2.JOG_SCRATCH_RPM,
+            RDJ2.JOG_SCRATCH_ALPHA,
+            RDJ2.JOG_SCRATCH_BETA,
+            RDJ2.JOG_SCRATCH_RAMP);
+    } else if (!this.jogTouchState && engine.isScratching(this.number)) {
+        engine.scratchDisable(this.number, RDJ2.JOG_SCRATCH_RAMP);
+    }
+};
+
+RDJ2.Deck.prototype.onJogSpin = function(channel, control, value) {
+    var currentJogMode =  this.jogModeSelector.getJogMode();
+    var jogDelta = RDJ2.getJogDeltaValue(value);
+
+    if (currentJogMode === RDJ2.JOGMODES.vinyl) {
+        engine.scratchTick(this.number, jogDelta);
+    } else if (currentJogMode === RDJ2.JOGMODES.fxdrywet) {
+        var currMixValue = engine.getParameter("[EffectRack1_EffectUnit" + this.number + "]", "mix");
+        engine.setParameter("[EffectRack1_EffectUnit" + this.number + "]", "mix", currMixValue + jogDelta / RDJ2.JOG_RESOLUTION);
+    } else if (currentJogMode === RDJ2.JOGMODES.search) {
+        var playPos = engine.getValue(this.group, "playposition");
+        if (undefined !== playPos) {
+            var seekPos = playPos + (jogDelta / (RDJ2.JOG_RESOLUTION * RDJ2.JOG_SEEK_REVOLUTIONS));
+            this.setValue("playposition", Math.max(0.0, Math.min(1.0, seekPos)));
+        }
+    } else if (currentJogMode === RDJ2.JOGMODES.trax) {
+        engine.setValue("[Library]", "MoveVertical", jogDelta);
```

Comment:
This is not necessary, is it? The Trax knob already does move and scroll, and for track selection you'd have to move to the Trax knob / Load buttons anyway?

FWIW, on my TerminalMix I mapped Shift+wheel turn to change the BPM by +-1 to cover BPM outside the pitch slider range (with decreased sensitivity to allow more precision)

Reply:
Changing the tracks star ratiing or track color could also be helpful additions.

Reply:
> This is not necessary, is it? The Trax knob already does move and scroll, and for track selection you'd have to move to the Trax knob / Load buttons anyway?

Not necessarily ;) Navigating a long tracklist is cumbersome as you have to do a lot of turns with the small Trax knob to get to the desired track and the feeling of the knob is not that great (it clicks at every digital step and feels cheap). So this is an enhancement to the track browsing. Browsing with the jog wheel is fast and smooth.

Reply:
Okay, so every wheel tick moves the selection by one line? This is indeed smoother than the emulated PageUp/Down signals sent by `ScrollVertical`

Reply:
I recently also introduced scrolling through the library with jog wheel after I saw that in Traktor. It is a great way to quickly navigate through long lists. Only applied this if the library is maximized though... 

https://github.com/mixxxdj/mixxx/blob/main/res/controllers/Denon-MC7000-scripts.js#L678